lib/p5.play.js

```diff
@@ -2189,246 +2123,230 @@ function Sprite(pInst, _x, _y, _w, _h) {
+        var newVel1 = dx1.mult(m1 * p5.Vector.sub(this.velocity, target.velocity).dot(dx1) / magnitude);
+        var newVel2 = dx2.mult(m2 * p5.Vector.sub(target.velocity, this.velocity).dot(dx2) / magnitude);
 
-                  }
+        this.velocity.sub(thisOldVelOnAxis).sub(newVel1.mult(this.restitution));
+        target.velocity.sub(targetOldVelOnAxis).sub(newVel2.mult(target.restitution));
+      }
+    }
 
-              }//end overlap
+    // Finally, for all collision types call the callback and record
+    // that collision occurred.
+    if (typeof callback === 'function') {
+      callback.call(this, this, target);
+    }
+    return true;
+  };
 
+  this._findDisplacement = function(target) {
+    // Multisample if tunneling occurs:
+    // If the direction between the two
+    // If the relative velocities of the two sprites is greater than the
+    // difference of their previous centers on the velocity axis,
+    // their centers might have crossed on an axis this frame which would permit
+    // tunneling.
+    // In that case, test interpolations between their last positions and their
+    // current positions, and check for tunneling that way.
+    var relativeVelocity = p5.Vector.sub(this.velocity, target.velocity);
+    var squareVelocity = relativeVelocity.magSq();
+    var centerSquareDistanceOnVelocityAxis = p5.Vector.project(
+      p5.Vector.sub(this.collider.center, target.collider.center),
+      relativeVelocity
+    ).magSq();
+
+    // If the collider centers are closer on the velocity axis than the
+    // velocity itself, we are close enough to have tunneled this frame.
+    // Use multisampling to catch collisions we might otherwise miss.
+    if (squareVelocity > centerSquareDistanceOnVelocityAxis) {
+      // Figure out how many samples we should take.
+      var radiusOnVelocityAxis = Math.max(
+        this.collider._getRadiusOnAxis(relativeVelocity),
+        target.collider._getRadiusOnAxis(relativeVelocity));
+      var timestep = radiusOnVelocityAxis / Math.sqrt(squareVelocity);
+      // If the objects are small enough to benefit from multisampling at this
+      // relative velocity
+      if (timestep < 1) {
+        // Move sprites back to previous positions
+        // (We jump through some hoops here to avoid creating too many new
+        //  vector objects)
+        var thisOriginalPosition = this.position.copy();
+        var targetOriginalPosition = target.position.copy();
+        this.position.set(this.previousPosition);
+        target.position.set(target.previousPosition);
+
+        // Scale deltas down to timestep-deltas
+        var thisDelta = p5.Vector.sub(thisOriginalPosition, this.previousPosition).mult(timestep);
+        var targetDelta = p5.Vector.sub(targetOriginalPosition, target.previousPosition).mult(timestep);
+
+        // Note: We don't have to check the original position, we can assume it's
+        // non-colliding (or it would have been handled on the last frame).
```

Comment:
Since we're never going to check the start-of-frame position in a non-tunneling case I think I'm fine with this.  I did realize at some point that start positions that were not overlapping on the last frame could be overlapping now due to rotation changes, etc - but those cases are so likely to be caught by the next subframe sample that I'm not sure anybody would notice.